### PR TITLE
ref(go.mod): oras v0.8.1

### PR DESCRIPTION
**Pull request by jdolitsky**
Upgrade to oras 0.8.0

Still 2 replace statements in go.mod needed, hoping to follow up soon to get rid of them completely...

### CODE REVIEW

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @jdolitsky

**Comment by bacongobbler** [comment]
Hey @hickeyma and @jdolitsky! I'm trying to gauge whether this should go in for 3.0.0 or not. Since OCI support is an experimental feature of Helm, I'm thinking this should go into 3.1.0. This is one of the rare exceptions where breaking changes can be rolled along as oras continues to iterate. I'm also concerned about merging in a change that seemingly downgrades the docker client libraries.

What are your thoughts on this?

**Comment by jdolitsky** [comment]
I'm happy to punt this. The whole oras 0.8 release came out of discussion w Martin about non-release version of containerd, which is only used in the experimental stuff.

Although it appears that `github.com/Azure/go-autorest` wasnt actually present in `go.sum`, does this mean it was in use? It looks like pkg `github.com/Azure/go-autorest/autorest` is still locked down:
```
github.com/Azure/go-autorest/autorest v0.9.0 h1:MRvx8gncNaXJqOoLmhNjUAKh33JJF8LyxPhomEtOsjs=
github.com/Azure/go-autorest/autorest v0.9.0/go.mod 
```

**Comment by hickeyma** [comment]
The reason behind this was because ORAS and as a consequence Helm was using unreleased `containerd` and `docker` modules. It is best practice to try and release with dependencies modules that are released. 

This also  has a knock on effect to apps using Helm as a client. An example is the Helm [2to3](https://github.com/helm/helm-2to3) plugin which needs to replace docker module as follows:
`github.com/docker/docker => github.com/docker/docker v0.0.0-20190731150326-928381b2215c`.

I am not sure about Azure's go-autorest client libraries. Why are they need by Helm?

**Comment by mattfarina** [comment]
My hesitation on this is that we are in a release candidate phase. That means we should only be merging in fixes for significant bugs to 3.0.0. We're past the point that dependency updates should go in on their own.

**Comment by hickeyma** [comment]
@mattfarina I agree with your response. It is probably too late in the day now that we are in the RC phase.

**Comment by mattfarina** [comment]
@jdolitsky Good catch on autorest.  That replace statement was part of the original setup for Go modules to support Kubernetes. Since that is no longer being imported this replace can be dropped as part of another PR.

**Comment by jdolitsky** [comment]
updated a few more things, this should be good to merge

File: go.mod [hunk 1]
```diff
@@ -4,89 +4,50 @@ go 1.13
...
+	github.com/Masterminds/vcs v1.13.1
 	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a
-	github.com/containerd/containerd v1.3.0
+	github.com/containerd/containerd v1.3.1
 	github.com/cyphar/filepath-securejoin v0.2.2
-	github.com/deislabs/oras v0.7.0
+	github.com/deislabs/oras v0.8.0
 	github.com/docker/distribution v2.7.1+incompatible
-	github.com/docker/docker v1.4.2-0.20181221150755-2cb26cfe9cbf
+	github.com/docker/docker v0.7.3-0.20190826074503-38ab9da00309
```
Comment (bacongobbler) [inline go.mod]: I'm curious why there appears to be a version downgrade for docker/docker. Do you know why this is the case?

File: go.mod [hunk 2]
```diff
@@ -4,89 +4,50 @@ go 1.13
...
 	k8s.io/cli-runtime v0.0.0-20191016114015-74ad18325ed5
 	k8s.io/client-go v0.0.0-20191016111102-bec269661e48
 	k8s.io/klog v1.0.0
-	k8s.io/kube-openapi v0.0.0-20190918143330-0270cf2f1c1d // indirect
 	k8s.io/kubectl v0.0.0-20191016120415-2ed914427d51
-	k8s.io/utils v0.0.0-20191010214722-8d271d903fe4 // indirect
 	sigs.k8s.io/yaml v1.1.0
 )
 
+// TODO: remove these replace statements which seem to be required for using github.com/deislabs/oras
```
Comment (bacongobbler) [inline go.mod]: were you able to figure out why these are required, or is this still a TODO item?

File: go.mod (hunk 1 above)
Comment (jdolitsky) [reply go.mod]: Its very confusing to me.. you will notice that there used to be a replace to the `v0.7.3` version which is now removed:
```
github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309
```

File: go.mod (hunk 2 above)
Comment (jdolitsky) [reply go.mod]: still TODO

File: go.mod (hunk 1 above)
Comment (povils) [reply go.mod]: based on the date this version is newer, no?(20181221150755 -> 20190826074503)

**Comment by mattfarina** [comment]
@jdolitsky Can you rebase or otherwise clean up the conflict?

**Comment by jdolitsky** [comment]
This is rebased and should be ready for re-review

File: go.mod [hunk 3]
```diff
@@ -4,79 +4,42 @@ go 1.13
...
-	github.com/Masterminds/vcs v1.13.0
-	github.com/Microsoft/go-winio v0.4.12 // indirect
-	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a
-	github.com/containerd/containerd v1.3.0
+	github.com/Masterminds/vcs v1.13.1
+	github.com/asaskevich/govalidator v0.0.0-20200108200545-475eaeb16496
+	github.com/containerd/containerd v1.3.2
 	github.com/cyphar/filepath-securejoin v0.2.2
-	github.com/deislabs/oras v0.7.0
+	github.com/deislabs/oras v0.8.1-0.20200203195554-169a5428f68f
```
Comment (mattfarina) [inline go.mod]: I see that `v0.8.1` is out the door. It is the latest commit on the ORAS repo, too. This references using a different commit in `169a5428f68f`. Is there a reason to not use `v0.8.1` itself?

Comment (jdolitsky) [reply go.mod]: whoops, just fixed

**Review by mattfarina — APPROVED** [review]
lgtm

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**